WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -1094,17 +1094,19 @@ private extension StatsPeriodStore {
     }
 
     func setAllFetchingStatus(_ status: StoreFetchingStatus) {
-        state.summaryStatus = status
-        state.summaryLikesStatus = status
-        state.topPostsAndPagesStatus = status
-        state.topReferrersStatus = status
-        state.topPublishedStatus = status
-        state.topClicksStatus = status
-        state.topAuthorsStatus = status
-        state.topSearchTermsStatus = status
-        state.topCountriesStatus = status
-        state.topVideosStatus = status
-        state.topFileDownloadsStatus = status
+        transaction { state in
```

Comment:
Nice find! It'll be a bit of a long method, but once #17926 has merged can we pull those changes in from trunk and wrap this section in a call to `if FeatureFlag.statsPerformanceImprovements.enabled`? We can do the previous behaviour in the `else`. Just in case we spot any issues with the change later, it'll help to locate and tweak or disable our various performance updates.

Reply:
> Nice find! It'll be a bit of a long method, but once #17926 has merged can we pull those changes in from trunk and wrap this section in a call to `if FeatureFlag.statsPerformanceImprovements.enabled`? We can do the previous behaviour in the `else`. Just in case we spot any issues with the change later, it'll help to locate and tweak or disable our various performance updates.

Hey frosty I've updated this now to use FeatureFlag 😄